### PR TITLE
Implement liveReload utils to simplify live reload implementations

### PR DESCRIPTION
**Pull request by JiriLojda**
### Motivation

Which issue does this fix? Fixes #`issue number`

If no issue exists, what is the fix or new feature? Were there any reasons to fix/implement things that are not obvious?

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

### CODE REVIEW

**Review by IvanKiral — CHANGES_REQUESTED** [review]

File: src/utils/liveReload.ts [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+const applyUpdateOnItemOptionallyAsync = <Elements extends IContentItemElements>(
+  item: IContentItem<Elements>,
+  update: InternalUpdateMessage,
+  resolveElementCodename: null | ((codename: string) => string)
+): OptionallyAsync<IContentItem<Elements>> => {
+  const shouldApplyOnThisItem =
+    item.system.codename === update.item.codename && item.system.language === update.variant.codename;
+
+  const resolveCodename = resolveElementCodename ?? ((c: string) => camelCasePropertyNameResolver('', c));
```
[the remark this answers is not in the document]
Comment (JiriLojda) [reply src/utils/liveReload.ts:49]: The SDK uses camelCase by default. When you don't configure it explicitly you will have camelCase codenames in `item.elements` keys in `res.data` of SDK responses.

File: src/utils/liveReload.ts [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+  switch (el1.type) {
+    case ElementType.Text:
+    case ElementType.Number:
+    case ElementType.UrlSlug:
+      return el1.value === el2.value;
+    case ElementType.MultipleChoice: {
+      const typedElement1 = el1 as Elements.MultipleChoiceElement;
+      const typedElement2 = el2 as Elements.MultipleChoiceElement;
+      return (
+        typedElement1.value.length === typedElement2.value.length &&
```
[the remark this answers is not in the document]
Comment (JiriLojda) [reply src/utils/liveReload.ts:226]: I am not sure what you mean.

[the remark this answers is not in the document]
Comment (IvanKiral) [reply src/utils/liveReload.ts:226]: oh sorry, I thought that multipleChoice has `mode` property. I have mistaken it for management api

File: src/utils/liveReload.ts [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+          (newElement) => [elementCodename, newElement] as const
+        );
+      }
+
+      if (element.type === ElementType.ModularContent || element.type === ElementType.RichText) {
+        const typedItemElement = element as Elements.LinkedItemsElement | Elements.RichTextElement;
+
+        return applyOnOptionallyAsync(
+          mergeOptionalAsyncs(
+            typedItemElement.linkedItems.map((i) => applyUpdateOnItemOptionallyAsync(i, update, resolveElementCodename))
```
[the remark this answers is not in the document]
Comment (JiriLojda) [reply src/utils/liveReload.ts]: This is a very good point. I updated the code to handle item cycles properly coupled with a few tests.

File: README.md [hunk 4]
```diff
@@ -461,17 +461,24 @@ application.
...
 // Initialize the SDK
 const sdk = KontentSmartLink.initialize({ ... });
 
 // Listen for updates and apply them to your application
 sdk.on(KontentSmartLink.Update, (data: IUpdateMessageData) => {
-  // Use this data to update your application state or UI as needed
+  // Use this data to update your application state or UI as needed e.g.:
+  setItems((items) => items.map(item => applyUpdateOnItem(item, data)));
+  // or
+  Promise.all(items.map(item => applyUpdateOnItemAndLoadLinkedItems(item, data, fetchItemsFromDeliveryApi)))
```
Comment (IvanKiral) [inline README.md:480]: don't we want to add below? 
I know its up to anybody how he want to get the items, but this example would save me 2 or 3 minutes till i figured out how to do it :D
```ts
const fetchItemsFromDeliveryApi = (items) => client.items().inFilter(system.codename, items).toAllPromise().then(res => res.data.items)

```

**Review by IvanKiral — DISMISSED** [review]

**Pull request merged**